File: w2d5/lib/transpose.rb
def my_transpose(matrix)
  raise "Not a matrix" unless matrix.all? { |row| row.count == matrix[0].count }
  return matrix if matrix.empty?
  transposed_matrix = []
  matrix[0].count.times do |col_i|
    new_row = Array.new
    matrix.each do |row|
      new_row << row[col_i]
    end
    transposed_matrix << new_row
  end
  transposed_matrix

end

File: W6/W6D1/transpose.js
var transpose = function(matrix) {
  var array = [];
  for(var i = 0; i < matrix[0].length; i++) {
    array.push(new Array(matrix.length));
  }

  for(var row = 0; row < matrix.length; row++) {
    for(var col = 0; col < matrix[row].length; col++) {
      array[col][row] = matrix[row][col];
    };
  }
  return array;
}